### DISCUSSION TITLE
Pass arguments to elastalert enhancements for reusability

### DISCUSSION THREAD

**Discussion by BenJeau** [jertel/elastalert2#841] [Q&A] [answered]
Hi, I would like to pass arguments/parameters to enhancements (https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html) to make them more reusable. For example, I have an enhancement that modifies a field to get the top level domain of a URL field, but I'd like to use it on multiple different fields without needing to create new enhancements. I don't think that is currently supported (if I'm wrong please let me know), I would propose a simple syntax change for them in the config files

```yaml
match_enhancements:
- "elastalert_modules.my_enhancements.MyEnhancement"
  - param1
  - param2
```

or

```yaml
match_enhancements:
- name: "elastalert_modules.my_enhancements.MyEnhancement"
  params:
  - param1
  - param2
```

and in the Python side, maybe the following:

```py
from elastalert.enhancements import BaseEnhancement

class MyEnhancement(BaseEnhancement):
    def process(self, match, param1, param2):
        pass
```

or 

```py
from elastalert.enhancements import BaseEnhancement

class MyEnhancement(BaseEnhancement):
    def process(self, match, *params):
        pass
```

**Comment by jertel** [comment, marked as answer]
Have you tried making the enhancement read the params from the rule dict?

```
match_enhancements:
- "mod.my_enhancements.MyEnhancement"
enhancement_foo: 123
enhancement_bar: abc
```

Inside your enhancement module:
```
foo = self.rule['enhancement_foo']
bar = self.rule['enhancement_bar']
```
